apps/coscrad-frontend/src/store/slices/resources/terms/hooks/use-loadable-terms.ts

```diff
@@ -1,10 +1,28 @@
-import { useLoadable } from '../../shared/hooks';
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { useAppDispatch } from '../../../../../app/hooks';
 import { selectLoadableTerms } from '../selectors';
 import { fetchTerms } from '../thunks';
 
-export const useLoadableTerms = () =>
-    useLoadable({
-        selector: selectLoadableTerms,
-        // @ts-expect-error fix this
-        fetchThunk: () => fetchTerms(null),
-    });
+export const useLoadableTerms = () => {
```

Comment:
It's interesting that we couldn't reuse the existing `useLoadable`. I do see the value in sharing that logic, but we have to find a robust way to do this. We don't want to do another big refactor down the road.